vent/logger/logger.py

```diff
@@ -0,0 +1,197 @@
+import time
+from typing import List
+import threading
+import numpy as np
+import copy
+from collections import deque
+import pdb
+import vent.io as io
+import logging
+import tables as pytb
+from datetime import datetime
+import os
+import string
+
+from vent.common.message import SensorValues, ControlValues, ControlSetting
+from vent.common.values import CONTROL, ValueName
+
+
+class DataSample(pytb.IsDescription):
+    """
+    Structure for the hdf5-table for data
+    """
+    timestamp    = pytb.Float64Col()    # current time of the measurement - has to be 64 bit
+    pressure     = pytb.UInt8Col()
+    flow_in      = pytb.UInt8Col()
+    flow_out     = pytb.UInt8Col()
+    cycle_number = pytb.UInt32Col()     # Max is 2147483647 Breath Cycles (~78 years)
+
+class ControlCommand(pytb.IsDescription):
+    """
+    Structure for the hdf5-table to store control commands
+    """
+    name      = pytb.StringCol(16)   # Control setting name
+    value     = pytb.Float64Col()    # double (double-precision)
+    min_value = pytb.Float64Col()    # double (double-precision)
+    max_value = pytb.Float64Col()    # double (double-precision)
+    timestamp = pytb.Float64Col()    # double (double-precision)
+
+
+class DataLogger:
+    """
+    Class for logging numerical respiration data and control settings.
+    Creates a hdf5 file with this general structure:
+        / root
+        |--- waveforms (group)
+        |    |--- time | pressure_data | volume | Cycle No.    --- TODO: WHAT TO SAVE?
+        |
+        |--- controls (group)                                  --- TODO: WHAT TO SAVE? DO THIS HERE?
+        |    |--- (time, controllsignal)
+        |
+
+    Public Methods:
+        open_logfile():                       Opens a log-file, creates one if necessary. 
+        close_logfile():                      Flushes, and closes the logfile.
+        store_waveform_data(SensorValues):    Takes data from SensorValues, but DOES NOT FLUSH 
+        store_controls():                     Store controls in the same file? TODO: Discuss 
+        flush_logfile():                      Flush the data into the file
+        check_size():                         Make sure that the files don't grow too big TODO: Implement
+
+    """
+
+    def __init__(self):
+
+        # If initialized, make a new file
+        today = datetime.today()
+        date_string = today.strftime("%Y-%m-%d-%H-%M")
+
+        # Make the log folder
+        if not os.path.exists('vent/logfiles'):
+            os.makedirs('vent/logfiles')
+        self.file = "vent/logfiles/" + date_string + "_controller_log.h5"
+
+        # Make sure that the file doesn't exist yet, if it does, append alphabet ('a','b','c'...)
+        # In rarely happens, but for Travis-tests, this is needed.
+        ls = list(string.ascii_lowercase)
+        ls = ls + list(string.ascii_uppercase)
```

Comment:
I would suggest using a number as a prefix instead of letter because you don't know how many logging could be.

Reply:
I liked the letters, because numbers might be confused with time. But I changed it anyway, my feelings aren't that strong on the subject